invisible_cities/reco/calib_sensors_functions.py

```diff
@@ -34,14 +34,34 @@ def mode(wfs, axis=0):
     positive waveforms.
     """
     def wf_mode(wf):
-        positive = wf >= 0
-        return np.bincount(wf[positive]).argmax() if np.count_nonzero(positive) else -1
+        positive = wf > 0
+        return np.bincount(wf[positive]).argmax() if np.count_nonzero(positive) else 0
     return np.apply_along_axis(wf_mode, axis, wfs).astype(float)
 
 
-def means  (wfs): return to_col_vector(np.mean  (wfs, axis=1))
-def medians(wfs): return to_col_vector(np.median(wfs, axis=1))
-def modes  (wfs): return to_col_vector(   mode  (wfs, axis=1))
+def median(wfs, axis=0):
+    """
+    The numpy median but with protection
+    so that we get the correct answer in case of 
+    zero suppressed data
+    """
+    mask_wfs = np.ma.masked_where(wfs == 0, wfs)
+    return np.ma.median(mask_wfs, axis=axis).filled(0)
+
+
+def mean(wfs, axis=0):
+    """
+    The numpy mean but with protection
+    so that we get the correct answer in case of 
+    zero suppressed data
+    """
+    mask_wfs = np.ma.masked_where(wfs == 0, wfs)
+    return np.ma.mean(mask_wfs, axis=axis).filled(0)
```

Comment:
`median` and `mean` are just *identical* thin wrappers around `np.ma.median` and `np.ma.mean`. As such the signal to noise ratio in them isn't all that great.

I would prefer something along the lines of 

    from functools import wraps

    def masked_version(fn):
        @wraps(fn)
        def proxy(wfs, *args, **kwds):
            maks_wfs = np.ma.masked_where(wfs == 0, wfs)
            return fn(maks_wfs, *args, **kwds).filled(0)
        proxy.__doc__ = "Masked version blah blah \n\n" + proxy.__doc__
        return proxy

    median = masked_version(np.ma.median)
    mean   = masked_version(np.ma.mean)

The exact mangling of the `__doc__` would require a few more seconds of thought.